src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs

```diff
@@ -357,101 +358,17 @@ public async Task<ResultWrapper<Keccak>> eth_sendRawTransaction(byte[] transacti
-            }
+        public ResultWrapper<UInt256?> eth_estimateGas(TransactionForRpc transactionCall, BlockParameter blockParameter) =>
+            new EstimateGasTransactionExecutor(_blockchainBridge, _blockFinder, _cancellationTokenTimeout, _rpcConfig)
+                .ExecuteTx(transactionCall, blockParameter);
 
-            transactionCall.From ??= Address.SystemUser;
-        }
-
-        public ResultWrapper<UInt256?> eth_estimateGas(TransactionForRpc transactionCall, BlockParameter blockParameter)
-        {
-            SearchResult<BlockHeader> searchResult = _blockFinder.SearchForHeader(blockParameter);
-            if (searchResult.IsError)
-            {
-                return ResultWrapper<UInt256?>.Fail(searchResult);
-            }
-
-            BlockHeader header = searchResult.Object;
-
-            if (!HasStateForBlock(header))
-            {
-                return ResultWrapper<UInt256?>.Fail($"No state available for block {header.Hash}",
-                    ErrorCodes.ResourceUnavailable);
-            }
-
-            return EstimateGas(transactionCall, header);
-        }
-
-        private ResultWrapper<UInt256?> EstimateGas(TransactionForRpc transactionCall, BlockHeader head)
-        {
-            // 2021-03-04 08:54:18.6489|DEBUG|101|Responded to ID 40, eth_estimateGas({
-            //     "from": "0x2b5ad5c4795c026514f8317c7a215e218dccd6cf",
-            //     "to": "0xa28afda14be5789564ae5fa03665c4180e3c680b",
-            //     "data": "0x25936984"
-            // }) 
-            
-            // 2021-03-04 08:54:18.6533|DEBUG|13|Responded to ID 41, eth_sendTransaction({
-            //     "gas": "0x1f815f1",
-            //     "from": "0x2b5ad5c4795c026514f8317c7a215e218dccd6cf",
-            //     "to": "0xa28afda14be5789564ae5fa03665c4180e3c680b",
-            //     "data": "0x25936984",
-            //     "gasPrice": "0x4a817c800"
-            // }) 
-            
-            FixCallTx(transactionCall);
-
-            // using CancellationTokenSource cancellationTokenSource = new(_cancellationTokenTimeout);
-            CancellationToken cancellationToken = CancellationToken.None;
-            BlockchainBridge.CallOutput result =
-                _blockchainBridge.EstimateGas(head, transactionCall.ToTransaction(_blockchainBridge.GetChainId()), cancellationToken);
-
-            if (result.Error == null)
-            {
-                return ResultWrapper<UInt256?>.Success((UInt256)result.GasSpent);
-            }
-
-            return ResultWrapper<UInt256?>.Fail(result.Error, result.InputError ? ErrorCodes.InvalidInput : ErrorCodes.InternalError);
-        }
+        public ResultWrapper<AccessListForRpc> eth_createAccessList(TransactionForRpc transactionCall, BlockParameter? blockParameter = null, bool optimize = true) =>
+            new CreateAccessListTransactionExecutor(_blockchainBridge, _blockFinder, _cancellationTokenTimeout, _rpcConfig, optimize)
```

Comment:
do not use => for complex calls

Reply:
if you move optimize to ExecuteTx then you can simply store CreateAccessListTxExecutor as a local field / local object